File: server/models/hashtagsonprofile.js
'use strict';
const {
  Model
} = require('sequelize');
module.exports = (sequelize, DataTypes) => {
  class hashtagsOnProfile extends Model {
    /**
     * Helper method for defining associations.
     * This method is not a part of Sequelize lifecycle.
     * The `models/index` file will call this method automatically.
     */
    static associate(models) {
      this.belongsTo(models.profiles);
      this.belongsTo(models.hashtags);
    }
  }
  hashtagsOnProfile.init({
    profileId: { type: DataTypes.INTEGER, allowNull: false },
    hashtagId: { type: DataTypes.INTEGER, allowNull: false },
    uuid: { type: DataTypes.UUID, allowNull: false, defaultValue: DataTypes.UUIDV4 }
  }, {
    sequelize,
    modelName: 'hashtagsOnProfile',
  });
  return hashtagsOnProfile;
};

File: lib/models/hashtag/hashtagsonprofile.dart
class HashtagsOnProfile {
  final int id;
  final int profileId;
  final int hashtagId;
  final DateTime createdAt;
  DateTime updatedAt;

  HashtagsOnProfile({
    required this.id,
    required this.profileId,
    required this.hashtagId,
    required this.createdAt,
    required this.updatedAt,
  });
}
